### PR TITLE
Slamdunk

### PR DESCRIPTION
**Pull request by t-neumann**
FOR CONTRIBUTOR:
* [ x ] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ x ] - License permits unrestricted use (educational + commercial)
* [ x ] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

FOR REVIEWER:
* [ ] .shed.yml file ok
    - [ ] Toolshed user `iuc` has access to associated toolshed repo(s)
* [ ] Indentation is correct (4 spaces)
* [ ] Tool version/build ok
* [ ] `<command/>`
  - [ ] Text parameters, input and output files `'single quoted'`
  - [ ] Use of `<![CDATA[ ... ]]>` tags
  - [ ] Parameters of type `text` or having `optional="true"` attribute are checked with `if str($param)` before being used
* [ ] Data parameters have a `format` attribute containing datatypes recognised by Galaxy
* [ ] Tests
  - [ ] Parameters are reasonably covered
  - [ ] Test files are appropriate
* [ ] Help
  - [ ] Valid restructuredText and uses `<![CDATA[ ... ]]>` tags
* [ ] Complies with other best practice in [Best Practices Doc](http://galaxy-iuc-standards.readthedocs.io/en/latest/best_practices/tool_xml.html)

### CODE REVIEW

File: tools/slamdunk/slamdunk.xml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    <requirements>
+        <requirement type="package" version="0.3.3">slamdunk</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+        slamdunk all -r "$input1" -b "$input2" -o "$output1" -rl 100 -mbq 27 -5 0 "$input3"
+    ]]></command>
+    <inputs>
+        <param type="data" name="input1" format="fa" />
+        <param type="data" name="input2" format="bed" />
+        <param type="data" name="input3" format="fq" />
```
[the remark this answers is not in the document]
Comment (mblue9) [reply tools/slamdunk/slamdunk.xml]: and also `fastqsanger.gz` please  (thanks for submitting this @t-neumann ! I just found out yesterday that I'll be needing to use this tool)

**Comment by bgruening** [comment]
Your test is currently failing because the test files do not match:

```
-- local_file
+++ history_data
@@ -1,2 +1,4 @@
+#slamdunk v0.3.3	3	sample info:	sample_0	0	NA	-1
+#annotation:	dataset_2.dat	bce6d51c7557e2f16db86a4b436c2e0b
 Chromosome	Start	End	Name	Length	Strand	ConversionRate	ReadsCPM	Tcontent	CoverageOnTs	ConversionsOnTs	ReadCount	TcReadCount	multimapCount	ConversionRateLower	ConversionRateUpper
-chr5	120498	122492	Actb	1994	+	0.0222222222222	666666.666667	445	90	2	8	4	0	-1.0	-1.0
+chr5	120498	122492	Actb	1994	+	0.015873015873	727272.727273	445	63	1	8	2	0	-1.0	-1.0
```

Also you removed some file in this PR, can you revert this. Also please use spaces instead of TAB in your tool definition. Thanks!

**Comment by t-neumann** [comment]
I have found that the section is not working for situations like this where I have an optional parameter -fb that if set will override others. With sections this one is always set it appears, which can be circumvented using conditionals.

I would rather not have a mix between sections and conditionals, or what is your take on this?

**Comment by t-neumann** [comment]
Ready for merge from my side. All conditionals replaced by sections.

File: tools/slamdunk/slamdunk.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (t-neumann) [reply tools/slamdunk/slamdunk.xml]: @mblue9 All done.

[the remark this answers is not in the document]
Comment (mblue9) [reply tools/slamdunk/slamdunk.xml]: Great, thanks!

**Comment by mblue9** [comment]
@t-neumann hope you don't mind I just added a commit to add using built-in fastas, as if the admin has set up those then the user doesn't have to upload a fasta. But I couldn't test it locally as there's no slamdunk bioconda package for Mac.

**Comment by mblue9** [comment]
Also I noticed a couple of minor things when using planemo serve e.g. this bit:

screen shot 2018-10-11 at 9 51 21 pm

- the user doesn't need to "call" `slamdunk all` that's just going to run automatically for them isn't it?
- the user in Galaxy doesn't specify the output directory (the files will appear in the history) so that might confuse them, maybe it could be removed?
- I don't think they'll need the files parameter either as they can't specify a samplesheet, they'll input the files from the history
- and the citation has "TODOs" written in it maybe you could just put your name for the moment?

**Comment by t-neumann** [comment]
@mblue9 Thanks for pointing this out. I made it more user-centered only explaining the necessary parameters to specify and some crucial parameter combinations.

@bgruening offered to take care of the citation.

**Comment by mblue9** [comment]
@t-neumann this looks great! I just made some small changes, hope they're ok for you. I'll merge when the tests pass. 

Nice work!! 👍

**Pull request merged**